nsqd/stats.go

```diff
@@ -145,6 +145,37 @@ func (n *NSQD) GetStats() []TopicStats {
 	return topics
 }
 
+func (n *NSQD) GetTopicStats(topic string) TopicStats {
+	n.RLock()
+	var realTopic *Topic
+	for _, t := range n.topicMap {
+		if topic == t.name {
+			realTopic = t
+			break
+		}
+	}
+	n.RUnlock()
+	realTopic.RLock()
+	realChannels := make([]*Channel, 0, len(realTopic.channelMap))
+	for _, c := range realTopic.channelMap {
+		realChannels = append(realChannels, c)
+	}
+	realTopic.RUnlock()
+	sort.Sort(ChannelsByName{realChannels})
+	channels := make([]ChannelStats, 0, len(realChannels))
+	for _, c := range realChannels {
+		c.RLock()
+		clients := make([]ClientStats, 0, len(c.clients))
+		for _, client := range c.clients {
```

Comment:
Another optimization that is possible here: For the Topic page we just need the count of clients, we don't need all their info. For the Channel page, we do need all the client info, but not for all Channels, just for the one.